frontend/controller/actions/utils.js

```diff
@@ -67,16 +67,20 @@ export const encryptedAction = (
       // are written
       const finished = enqueueDeferredPromise('encrypted-action')
 
+      let retainFailed = false
       try {
         // Writing to a contract requires being subscribed to it
         // Since we're only interested in writing and we don't know whether
         // we're subscribed or should be, we use an ephemeral retain here that
         // is undone at the end in a finally block.
-        await sbp('chelonia/contract/retain', contractID, { ephemeral: true })
+        await sbp('chelonia/contract/retain', contractID, { ephemeral: true }).catch(e => {
+          retainFailed = true
+          throw e
+        })
```

Comment:
There are dozens of other places in the project where we call `'chelonia/contract/retain'`, and yet none of them use this `retainFailed` check... why here?

Reply:
Because in those other places the retain is outside of the `try` block, as it should be. Here, it's inside the `try` block because otherwise everything inside of the `catch` needs to be replicated.

The goal is, in all situations, is that `release` should only be called on successful `retain` calls.